### DISCUSSION TITLE
Reducing visual noise 👁

### DISCUSSION THREAD

**Discussion by neon-age** [neon-age/Smart-Hierarchy#16] [Ideas]
So I've tried to set MeshFilter as a prioritized type over MeshRenderer, and it started to look so much easier for the eye!

![image](https://user-images.githubusercontent.com/29812914/107552671-667cf180-6c06-11eb-8e19-3c77d7d971c2.png)

And now I'm thinking to add some features to help reduce this visual noise created by some of the icons:
- "Prefer/Ignore" enum for Types Priority.
- Custom Icons for Types (ex. SpriteRenderer with Sprite icon).
- Support per-gameobject "Gizmo" icon.

Another idea that comes to my mind is "Compound Components (?)"
![image](https://user-images.githubusercontent.com/29812914/107563340-85ce4b80-6c13-11eb-917e-c2470abe889e.png)
Not sure if it's a good idea, as it might create confusion/distraction or result in even more noise..

What do you think?

**Comment by Thane5** [comment]
I would be careful with how you display those compound objects, vertical lines like these are often used to better visualize the hierarchy between objects (Blender has this for a while now).
